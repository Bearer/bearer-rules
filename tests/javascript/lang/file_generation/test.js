const { createInvoker, getEnvironment } = require("../../../helper.js")
const { ruleId, ruleFile, testBase } = getEnvironment(__dirname)

describe(ruleId, () => {
  const invoke = createInvoker(ruleId, ruleFile, testBase)
  
  test("file_generation", ({ expect }) => {
    const testCase = "file_generation.js"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  
})