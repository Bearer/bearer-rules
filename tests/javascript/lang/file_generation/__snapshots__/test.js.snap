// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`javascript_lang_file_generation file_generation 1`] = `
"{
  "high": [
    {
      "cwe_ids": [
        "313"
      ],
      "id": "javascript_lang_file_generation",
      "title": "Sensitive data detected as part of a dynamic file generation.",
      "description": "## Description\\n\\nIt is not uncommon to generate logs, backups, or data exports to static file formats. This rule checks if code exists to write sensitive data to static files.\\n\\n## Remediations\\n\\nAvoid writing sensitive data to logs, backups, or exports whenever possible. Instead obfuscate and/or filter the data to exclude sensitive information.\\n\\n<!--\\n## Resources\\nComing soon.\\n-->\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/javascript_lang_file_generation",
      "line_number": 7,
      "full_filename": "/tmp/bearer-scan/file_generation.js",
      "filename": ".",
      "data_type": {
        "category_uuid": "14124881-6b92-4fc5-8005-ea7c1c09592e",
        "name": "Firstname"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 7,
        "end": 7,
        "column": {
          "start": 35,
          "end": 49
        }
      },
      "sink": {
        "start": 15,
        "end": 18,
        "column": {
          "start": 1,
          "end": 3
        },
        "content": "fs.writeFile(\\"data.csv\\", JSON.stringify(users), \\"utf-8\\", (err) => {\\n  if (err) console.log(err)\\n  else console.log(\\"Data saved\\")\\n})"
      },
      "parent_line_number": 15,
      "snippet": "fs.writeFile(\\"data.csv\\", JSON.stringify(users), \\"utf-8\\", (err) => {\\n  if (err) console.log(err)\\n  else console.log(\\"Data saved\\")\\n})",
      "fingerprint": "7162a96ee591e4689c1fa24bfcc02fd5_0",
      "old_fingerprint": "e07392ef7687a29685f9b9f7fd673469_0",
      "code_extract": "fs.writeFile(\\"data.csv\\", JSON.stringify(users), \\"utf-8\\", (err) => {\\n  if (err) console.log(err)\\n  else console.log(\\"Data saved\\")\\n})"
    },
    {
      "cwe_ids": [
        "313"
      ],
      "id": "javascript_lang_file_generation",
      "title": "Sensitive data detected as part of a dynamic file generation.",
      "description": "## Description\\n\\nIt is not uncommon to generate logs, backups, or data exports to static file formats. This rule checks if code exists to write sensitive data to static files.\\n\\n## Remediations\\n\\nAvoid writing sensitive data to logs, backups, or exports whenever possible. Instead obfuscate and/or filter the data to exclude sensitive information.\\n\\n<!--\\n## Resources\\nComing soon.\\n-->\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/javascript_lang_file_generation",
      "line_number": 9,
      "full_filename": "/tmp/bearer-scan/file_generation.js",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 9,
        "end": 9,
        "column": {
          "start": 10,
          "end": 20
        }
      },
      "sink": {
        "start": 15,
        "end": 18,
        "column": {
          "start": 1,
          "end": 3
        },
        "content": "fs.writeFile(\\"data.csv\\", JSON.stringify(users), \\"utf-8\\", (err) => {\\n  if (err) console.log(err)\\n  else console.log(\\"Data saved\\")\\n})"
      },
      "parent_line_number": 15,
      "snippet": "fs.writeFile(\\"data.csv\\", JSON.stringify(users), \\"utf-8\\", (err) => {\\n  if (err) console.log(err)\\n  else console.log(\\"Data saved\\")\\n})",
      "fingerprint": "7162a96ee591e4689c1fa24bfcc02fd5_2",
      "old_fingerprint": "e07392ef7687a29685f9b9f7fd673469_2",
      "code_extract": "fs.writeFile(\\"data.csv\\", JSON.stringify(users), \\"utf-8\\", (err) => {\\n  if (err) console.log(err)\\n  else console.log(\\"Data saved\\")\\n})"
    },
    {
      "cwe_ids": [
        "313"
      ],
      "id": "javascript_lang_file_generation",
      "title": "Sensitive data detected as part of a dynamic file generation.",
      "description": "## Description\\n\\nIt is not uncommon to generate logs, backups, or data exports to static file formats. This rule checks if code exists to write sensitive data to static files.\\n\\n## Remediations\\n\\nAvoid writing sensitive data to logs, backups, or exports whenever possible. Instead obfuscate and/or filter the data to exclude sensitive information.\\n\\n<!--\\n## Resources\\nComing soon.\\n-->\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/javascript_lang_file_generation",
      "line_number": 10,
      "full_filename": "/tmp/bearer-scan/file_generation.js",
      "filename": ".",
      "data_type": {
        "category_uuid": "14124881-6b92-4fc5-8005-ea7c1c09592e",
        "name": "Firstname"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 10,
        "end": 10,
        "column": {
          "start": 15,
          "end": 29
        }
      },
      "sink": {
        "start": 15,
        "end": 18,
        "column": {
          "start": 1,
          "end": 3
        },
        "content": "fs.writeFile(\\"data.csv\\", JSON.stringify(users), \\"utf-8\\", (err) => {\\n  if (err) console.log(err)\\n  else console.log(\\"Data saved\\")\\n})"
      },
      "parent_line_number": 15,
      "snippet": "fs.writeFile(\\"data.csv\\", JSON.stringify(users), \\"utf-8\\", (err) => {\\n  if (err) console.log(err)\\n  else console.log(\\"Data saved\\")\\n})",
      "fingerprint": "7162a96ee591e4689c1fa24bfcc02fd5_3",
      "old_fingerprint": "e07392ef7687a29685f9b9f7fd673469_3",
      "code_extract": "fs.writeFile(\\"data.csv\\", JSON.stringify(users), \\"utf-8\\", (err) => {\\n  if (err) console.log(err)\\n  else console.log(\\"Data saved\\")\\n})"
    }
  ]
}"
`;
